File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/constantes/TipoFeriadoEnum.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.constantes;

public enum TipoFeriadoEnum {
    FERIADO("Feriado", "F"),
    PONTO_FACULTATIVO("Ponto Facultativo", "P"),
    BANCARIO("Feriado Exclusivamente Banc\u00e1rio", "B");

    private String nome;
    private String valor;

    private TipoFeriadoEnum(String nome, String valor) {
        this.nome = nome;
        this.valor = valor;
    }

    public String getNome() {
        return this.nome;
    }

    public String getValor() {
        return this.valor;
    }
}

File: src/lib/pjecalc/core/constantes/tipo-feriado-enum.ts
/**
 * PJe-Calc v2.15.1 — TipoFeriadoEnum
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.constantes.TipoFeriadoEnum
 *
 * Usado no cadastro de feriados (Tabela Feriado) e na apuração de
 * cartão de ponto para classificar dias não-úteis.
 *
 * Ref Java: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/constantes/TipoFeriadoEnum.java
 */

export enum TipoFeriadoEnum {
  FERIADO = 'F',
  PONTO_FACULTATIVO = 'P',
  /** Feriado exclusivamente bancário (não afeta trabalhista comum). */
  BANCARIO = 'B',
}

export const TipoFeriadoEnumNomes: Record<TipoFeriadoEnum, string> = {
  [TipoFeriadoEnum.FERIADO]: 'Feriado',
  [TipoFeriadoEnum.PONTO_FACULTATIVO]: 'Ponto Facultativo',
  [TipoFeriadoEnum.BANCARIO]: 'Feriado Exclusivamente Bancário',
};
